### DISCUSSION TITLE
How to Assign BT ROR in IKAWA Home implementation

### DISCUSSION THREAD

**Discussion by drgarys** [artisan-roaster-scope/artisan#1513] [Q&A]
Hello:

I’ve successfully installed Artisan for the IKAWA Home. Artisan 2.10.2 is able to pick up the programmed IKAWA data through Bluetooth. But to get true BT and exhaust temp, I’ve installed thermocouples and connected them to my MacBook Air M2 running MacOS Sonoma 14.3.1 via an Amprobe TMD-56 data logger connected to a USB port. 

(For anyone who isn’t using this, you load the roast profile to the IKAWA Home with your smart phone via Bluetooth. Then you exit the IKAWA Home app so that your computer can track the roast via Artisan and Bluetooth. You hit the Start button on the roaster to initiate the pre-programmed roast. You can’t run Bluetooth simultaneously on the phone and computer. ) 

I’ve got the above working but am unable to assign a BT ROR because it seems that Artisan defaults to the IKAWA Home meter, i.e., the roaster’s Bluetooth output for BT and ET. However, what Artisan is recording is inlet temp, not bean temp. At least I think that’s what I’m seeing on my Config screens. I may be mis-reading things. But here are screen shots of my settings. What do I need to do differently to see BT ROR?

<img width="941" alt="Screenshot 2024-03-13 at 10 44 21 PM" src="https://github.com/artisan-roaster-scope/artisan/assets/48734972/4a7327c6-67fc-4c74-b7a6-bc427842ef49">

<img width="1461" alt="Screenshot 2024-03-13 at 10 45 15 PM" src="https://github.com/artisan-roaster-scope/artisan/assets/48734972/3a4565dd-814b-4e8c-9f20-6013fb15dbd7">

My Artisan roast meter looks like this. Since then I’ve swapped the BT and ET probes where they plug in to the data logger. Showing events such as first crack start on the BT curve will be the next setting to solve.

![Hasan Yemen BT Exhaust Reversed](https://github.com/artisan-roaster-scope/artisan/assets/48734972/c96fa860-e799-47cf-9d84-f58f630c8f50)

**Comment by poundy** [comment]
I don't really understand your question. This sample shows us a BT line. To me that means you have no problem? 
You'd need to save a copy of your settings out and then add ".txt" to the end of the extension and attach the file here on github (not via an email response)

**Reply by drgarys** [reply]
If you look at the Artisan recording of my roast, the dark blue line is BT measured via my extra thermocouple added to the roasting chamber. I am unable to generate a BT ROR curve from that.

Here’s another screen. It shows that I have selected BT ROR but it doesn’t display.

<img width="649" alt="Screenshot 2024-03-14 at 2 26 26 AM" src="https://github.com/artisan-roaster-scope/artisan/assets/48734972/bb2bca36-6155-4d29-a79c-a585dc5dbacc">


My main question is whether there is a way for me to display a BT ROR for the actual BT being measured by the thermocouple I have added to the roasting chamber. I don’t see a way to do it.

Here are my settings: 

artisan-settings-Ikawa031424.aset.txt](https://github.com/artisan-roaster-scope/artisan/files/14600715/artisan-settings-Ikawa031424.aset.txt

**Reply by roasterdave** [reply]
Try using a capital 'Y' in the symbolic equations.

![Screenshot - 3_14_2024 , 8_56_55 AM](https://github.com/artisan-roaster-scope/artisan/assets/30009312/242f17d2-d542-4d8f-ac6d-a947167b8e49)

**Reply by MAKOMO** [reply]
The Symbolic Assignments Dave suggests will map the signals of your extra device to the main device, overwriting the IKAWAs BT and ET channels (which actually reports Exhaust and Inlet).

To additionally log those original IKAWA temperature channels you can add another extra device of type IKAWA. Note that the main device needs to remain IKAWA (despite all its data is overwritten by that symbolic assignment) as without that Artisan will not connect via BT.

For a lack of a device we cannot verify that this works. Sorry. Please report any success on this here.

**Comment by drgarys** [comment]
Thanks everybody! 

Dave’s solution of capitalizing the Y value worked with the extra step of confirming that I wanted to overwrite the ET/BT. Should that be a value that can be entered upper or lower case and still work? 

The events I marked now show on the BT curve (the profile was recorded before I swapped thermocouples in the Amprobe data logger). The delta ET/BT curves are very exaggerated. I wonder if there is a way to make that more readable?

![Hasan Yemen corrected](https://github.com/artisan-roaster-scope/artisan/assets/48734972/dca36d5b-5cb4-4ac6-8908-cd9e1cb2d68f)

I was able to make it a bit more readable by configuring Axes but wonder if there is more that I can do to make it readable?

<img width="776" alt="Screenshot 2024-03-14 at 10 41 11 AM" src="https://github.com/artisan-roaster-scope/artisan/assets/48734972/816ae202-5c57-44e8-bdcd-4afcee9e8143">

![Hasan Yemen corr 2](https://github.com/artisan-roaster-scope/artisan/assets/48734972/62e1c28c-06ac-44c4-a05a-1d0ba7b56379)

**Reply by roasterdave** [reply]
> I wonder if there is a way to make that more readable?

You can apply some filtering. Config>> Curves>> Filters.  Remember that BT RoR is the slope of the BT signal calculated every sample time. Your BT signal is not very smooth thus the RoR goes up and down as a result of every BT wiggle.  You could try smoothing BT a little.  Set Smooth Curves to 1 or 2 to smooth BT (and ET) and see if it improves.  Too much smoothing will mask the true signal.  The curve may look pretty but the information is gone.   Use smoothing sparingly.  I would not increase your RoR Delta Span nor RoR Smoothing.  

You may find noticeable benefit from slowing your sample rate.  First try 2 seconds and then for comparison try 4 seconds to see if the curves work better for you.

There is a very good explanation of how sampling, smoothing and rate of rise interact in this [Artisan Blog](https://artisan-roasterscope.blogspot.com/2014/01/sampling-interval-smoothing-and-rate-of.html) post.

> ...capitalizing the Y value... Should that be a value that can be entered upper or lower case and still work?

Making symbolic equations case insensitive is more involved than you might think.  It is not currently planned so for now they remain case sensitive.  You are welcome to create a [Feature Request](https://github.com/artisan-roaster-scope/artisan/issues/new?assignees=&labels=&projects=&template=feature_request.md&title=).
